stm/stm.ml

```diff
@@ -1341,6 +1313,46 @@ end (* }}} *)
 
 let get_prev_proof = Backtrack.get_prev_proof
 
+(* Wrapper for Vernacentries.interp to set the feedback id *)
+(* It is currently called 19 times, this number should be certainly
+   reduced... *)
+let stm_vernac_interp ?proof ?route id st { verbose; loc; expr } : Vernacstate.t =
+  (* The Stm will gain the capability to interpret commmads affecting
+     the whole document state, such as backtrack, etc... so we start
+     to design the stm command interpreter now *)
+  set_id_for_feedback ?route dummy_doc id;
+  Aux_file.record_in_aux_set_at ?loc ();
+  (* We need to check if a command should be filtered from
+   * vernac_entries, as it cannot handle it. This should go away in
+   * future refactorings.
+  *)
+  let is_filtered_command = function
+    | VernacResetName _ | VernacResetInitial | VernacBack _
+    | VernacBackTo _ | VernacRestart | VernacUndo _ | VernacUndoTo _
+    | VernacAbortAll | VernacAbort _ -> true
+    | _ -> false
+  in
+  let aux_interp st expr =
+    (* XXX unsupported attributes *)
+    let cmd = Vernacprop.under_control expr in
+    if is_filtered_command cmd then
+      (stm_pperr_endline Pp.(fun () -> str "ignoring " ++ Ppvernac.pr_vernac expr); st)
+    else
+      match cmd with
+      | VernacShow ShowScript -> ShowScript.show_script (); st (* XX we are ignoring control here *)
+      | _ ->
+        stm_pperr_endline Pp.(fun () -> str "interpreting " ++ Ppvernac.pr_vernac expr);
+        let doc = get_doc 0 in
+        let dproof = get_prev_proof ~doc (get_current_state ~doc) in
+        try Vernacentries.interp ?verbosely:(Some verbose) ?proof ~st dproof (CAst.make ?loc expr)
```

Comment:
The fact that the previous proof needs to be added to a core function highlights the problem with overloading the base API which was already sketchy (`Show` I'm looking at you)

It is not clear to me if we should keep adding kludges or the moment for UI developers has arrived: they either adopt a modern protocol or they renounce to advanced functionality.